File: Python/Leetcode Medium/150EvaluateReversePolishNotation.py
#https://leetcode.com/problems/evaluate-reverse-polish-notation/submissions/1491625252/
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for char in tokens:
            if char in "+-/*":
                r = stack.pop()
                l = stack.pop()
                if char == "*":
                    stack.append(l * r)
                elif char == "/":
                    stack.append(int(float(l) / r))
                elif char == "+":
                    stack.append(l + r)
                else:
                    stack.append(l - r)
            else:
                stack.append(int(char))
        return stack.pop()

File: C++/Leetcode Medium/150EvaluateReversePolishNotation.cpp
#include <vector>
using namespace std;
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        for (int i = 0; i < tokens.size(); ++i) {
            string curr = tokens[i];
            if (curr == "+" || curr == "-" || curr == "*" || curr == "/") {
                int val1 = stack.back();
                stack.pop_back();
                int val2 = stack.back();
                stack.pop_back();
                if (curr == "+") stack.push_back(val2 + val1);
                else if (curr == "-") stack.push_back(val2 - val1);
                else if (curr == "/") stack.push_back(val2 / val1);
                else if (curr == "*") stack.push_back(val2 * val1);
            }
            else {
                stack.push_back(stoi(curr));
            }
        }
        return stack.back();
    }
};